src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -75,37 +72,40 @@ static int
 get_driindex(void)
 {
     int available = -ENOENT;
-    char device[512] = KMSDRM_DRI_PATH;
+    char device[32];
     int drm_fd;
     int i;
     int devindex = -1;
-    DIR *folder = opendir(device);
+    DIR *folder;
+
+    SDL_strlcpy(device, kmsdrm_dri_path, kmsdrm_dri_pathsize + 1);
+    folder = opendir(device);
     if (!folder) {
         SDL_SetError("Failed to open directory '%s'", device);
         return -ENOENT;
     }
 
-    strncpy(device + KMSDRM_DRI_PATHSIZE, KMSDRM_DRI_DEVNAME,
-            KMSDRM_DRI_DEVNAMESIZE + 1);
+    SDL_strlcpy(device + kmsdrm_dri_pathsize, kmsdrm_dri_devname,
+                kmsdrm_dri_devnamesize + 1);
     for (struct dirent *res; (res = readdir(folder));) {
-        if (SDL_memcmp(res->d_name, KMSDRM_DRI_DEVNAME,
-                       KMSDRM_DRI_DEVNAMESIZE) == 0) {
-            strncpy(device + KMSDRM_DRI_PATHSIZE + KMSDRM_DRI_DEVNAMESIZE,
-                    res->d_name + KMSDRM_DRI_DEVNAMESIZE, 6);
+        if (SDL_memcmp(res->d_name, kmsdrm_dri_devname,
+                       kmsdrm_dri_devnamesize) == 0) {
+            SDL_strlcpy(device + kmsdrm_dri_pathsize + kmsdrm_dri_devnamesize,
+                        res->d_name + kmsdrm_dri_devnamesize, 6);
```

Comment:
You shouldn't hardcode 6 here, instead use `sizeof(device) - kmsdrm_dri_pathsize - kmsdrm_dri_devnamesize`
I can't tell off the top of my head whether there will be truncation here. Are we just copying a digit?

Reply:
Yup, just some digits